src/index.js

```diff
@@ -19,10 +19,25 @@ function getEnsAddress(networkId) {
   }
 }
 
+/**
+ * Get Resolver Contract
+ * // todo is Provider here meant to be provider from web3-core ? (may need to upgrade web3 library)
```

Comment:
The `provider` is used at `ethers.Contract` so whatever the type definition ethers.io uses. Ethers.js is written in Type script so probably you can import type from there?